aten/src/ATen/native/sparse/SparseTensor.cpp

```diff
@@ -741,6 +745,29 @@ SparseTensor sparse_mask(const Tensor& t, const SparseTensor& mask) {
     return mask.clone().to(t.device(), t.scalar_type());
   }
 
+  if (t.layout() == at::kSparse) {
+    auto intersection = at::empty({0}, t.options());
+    sparse_mask_intersection_out_stub(intersection.device().type(), intersection, t, mask);
+
+    if (intersection._nnz() == 0) {
+      return mask.clone().to(t.device(), t.scalar_type());
+    }
+
+    // TODO: once union kernels are fast, reimplement with something along the lines of
+    // return intersection + zeros_like(mask)
+    const auto union_indices = at::cat(
```

Comment:
All this to respect the semantics of the result inheriting the indices of the mask.